### PR TITLE
Bossens-Heckbert smoothing

### LINKED ISSUE #85: potentially better force function
* I've been experimenting with this force function from the paper attached Eq. 3.9 
* It indeed converges faster and produces higher mesh qualities than the default DistMesh one. Screen Shot 2021-09-05 at 12 36 56 PM. 
* Perhaps it would be useful to implement here?

[imp_distmesh2d.pdf](https://github.com/nschloe/dmsh/files/7112006/imp_distmesh2d.pdf)

### PR DESCRIPTION
**Pull request by nschloe**
Fixes #85.

### CODE REVIEW

File: src/dmsh/main.py [hunk 1]
```diff
@@ -263,11 +263,11 @@ def distmesh_smoothing(
         # Evaluate element sizes at edge midpoints
         edge_midpoints = (mesh.points[edges[:, 1]] + mesh.points[edges[:, 0]]) / 2
         p = target_edge_size_function(edge_midpoints.T)
-        desired_lengths = (
+        target_lengths = (
             f_scale * p * np.sqrt(np.dot(edge_lengths, edge_lengths) / np.dot(p, p))
```
Comment (krober10nd) [inline src/dmsh/main.py:267]: for the scaling of the `target_lengths` I like to use the ratio of the medians as it represents the inflation factor better in domains with large disparities in mesh size better than the ratio of squares.

```
# Bossen-Heckbert
def _compute_forces(p, t, fh, min_edge_length, L0mult):
    """Compute the forces on each edge based on the sizing function"""
    N = p.shape[0]
    bars = _get_bars(t)
    barvec = p[bars[:, 0]] - p[bars[:, 1]]  # List of bar vectors
    L = np.sqrt((barvec ** 2).sum(1))  # L = Bar lengths
    L[L == 0] = np.finfo(float).eps
    hbars = fh(p[bars].sum(1) / 2)
    # this line is what I'm referring to in my comment above.
    L0 = hbars * L0mult * (np.nanmedian(L) / np.nanmedian(hbars))
    LN = L / L0
    F = (1 - LN ** 4) * np.exp(-(LN ** 4)) / LN
    Fvec = (
        F[:, None] / LN[:, None].dot(np.ones((1, 2))) * barvec
    )  # Bar forces (x,y components)
    Ftot = _dense(
        bars[:, [0] * 2 + [1] * 2],
        np.repeat([list(range(2)) * 2], len(F), axis=0),
        np.hstack((Fvec, -Fvec)),
        shape=(N, 2),
    )
    return Ftot
```

Comment (nschloe) [reply src/dmsh/main.py:270]: Interesting. I'll add a separate issue for this.

Comment (nschloe) [reply src/dmsh/main.py:270]: Is this code in SeismicMesh?

Comment (krober10nd) [reply src/dmsh/main.py:270]: Yea, I was doing it offline. Now I just uploaded some quick implementation in there in https://github.com/krober10nd/SeismicMesh/pull/226

Ideally it should go in maybe as an option. I haven't explored 3D yet.

**Pull request merged**